tests/test_s3.py

```diff
@@ -2093,6 +2093,327 @@ def test_delete_bucket_notification(self):
+                    "Effect": "Allow",
+                    "Resource": "arn:aws:s3:::cross-account-invalid/*",
+                    "Principal": "*"
+                }
+            ]
+        }
+        client.put_bucket_policy(Bucket=bnames[1], Policy=json.dumps(policy))
+        p = self.load_policy({
+            'name': 's3_cross_account_vpce',
+            'resource': 's3',
+            'filters': [
+                {'type': 'cross-account',
+                 'whitelist': ['vpce-12345678']}
+            ]
+        }, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(resources[0]['Name'], 'cross-account-invalid')
+
+    def test_s3_cross_account_vpc(self):
+        self.patch(s3.S3, 'executor_factory', MainThreadExecutor)
+        self.patch(
+            s3.MissingPolicyStatementFilter, 'executor_factory',
+            MainThreadExecutor)
+        self.patch(s3, 'S3_AUGMENT_TABLE', [
+            ('get_bucket_policy', 'Policy', None, 'Policy'), ])
+        session_factory = self.replay_flight_data(
+            'test_s3_cross_account_vpc')
+        bnames = ['cross-account-valid', 'cross-account-invalid']
+        session = session_factory(region='us-east-1')
+        client = session.client('s3')
+        for b in bnames:
+            client.create_bucket(Bucket=b)
+            self.addCleanup(destroyBucket, client, b)
+        policy = {
+          "Id": "Policy1510863841605",
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Sid": "Stmt1510843305330",
+              "Action": [
+                "s3:PutObject"
+              ],
+              "Effect": "Allow",
+              "Resource": "arn:aws:s3:::cross-account-valid/*",
+              "Condition": {
+                "StringLike": {
+                  "aws:SourceVpc": "vpc-12345678"
+                }
+              },
+              "Principal": "*"
+            }
+          ]
+        }
+        client.put_bucket_policy(Bucket=bnames[0], Policy=json.dumps(policy))
+        policy = {
+            "Id": "Policy1510863841605",
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "Stmt1510843305330",
+                    "Action": [
+                        "s3:PutObject"
+                    ],
+                    "Effect": "Allow",
+                    "Resource": "arn:aws:s3:::cross-account-invalid/*",
+                    "Principal": "*"
+                }
+            ]
+        }
+        client.put_bucket_policy(Bucket=bnames[1], Policy=json.dumps(policy))
+        p = self.load_policy({
+            'name': 's3_cross_account_vpc',
+            'resource': 's3',
+            'filters': [
+                {'type': 'cross-account',
+                 'whitelist': ['vpc-12345678']}
+            ]
+        }, session_factory=session_factory)
+        resources = p.run()
+        self.assertEqual(resources[0]['Name'], 'cross-account-invalid')
```

Comment:
Same here and the rest of these.  Need to check that only the required number of resources came back.